sympy/functions/elementary/integers.py

```diff
@@ -131,14 +131,23 @@ def _eval_number(cls, arg):
         if arg.is_NumberSymbol:
             return arg.approximation_interval(Integer)[0]
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        if arg0.is_integer:
+            return arg0
```

Comment:
The result may be either `arg0` or `arg0 - 1` depending on which side the expansion is taken. For example,
```
In [8]: floor(-x**2).nseries(x, 0, 1)                                           
Out[8]: -1
```
I would consider usting `self._eval_nseries` for all finite values of `arg0`.

Reply:
Is -1 the expected output here? Shouldn't it be 0?

Reply:
The leading term should be the same as the first (or only) term of the series expansion, hence -1 in this case

Reply:
Right, I think this has been resolved by the latest commit